packages/docker/src/release/version-utils.ts

```diff
@@ -0,0 +1,131 @@
+import { execSync } from 'child_process';
+import { writeFileSync, mkdirSync } from 'fs';
+import { dirname, join } from 'path';
+import { prompt } from 'enquirer';
+import type { ProjectGraphProjectNode } from '@nx/devkit';
+import type { FinalConfigForProject } from 'nx/src/command-line/release/version/release-group-processor';
+import { interpolateVersionPattern } from './version-pattern-utils';
+
+const DEFAULT_VERSION_SCHEMES = {
+  production: '{currentDate|YYMM.DD}.{shortCommitSha}',
+  hotfix: '{currentDate|YYMM.DD}.{shortCommitSha}-hotfix',
+};
+
+export const getDockerVersionPath = (
+  workspaceRoot: string,
+  projectRoot: string
+) => {
+  return join(workspaceRoot, 'tmp', projectRoot, '.docker-version');
+};
+
+export async function handleDockerVersion(
+  workspaceRoot: string,
+  projectGraphNode: ProjectGraphProjectNode,
+  finalConfigForProject: FinalConfigForProject,
+  dockerVersionScheme?: string
+) {
+  const availableVersionSchemes =
+    finalConfigForProject.dockerOptions.versionSchemes ??
+    DEFAULT_VERSION_SCHEMES;
+  const versionScheme =
+    dockerVersionScheme && dockerVersionScheme in availableVersionSchemes
+      ? dockerVersionScheme
+      : await promptForNewVersion(
+          availableVersionSchemes,
+          projectGraphNode.name
+        );
+  const newVersion = calculateNewVersion(
+    projectGraphNode.name,
+    versionScheme,
+    availableVersionSchemes
+  );
+  const logs = updateProjectVersion(
+    newVersion,
+    workspaceRoot,
+    projectGraphNode.data.root,
+    finalConfigForProject.dockerOptions.repositoryName,
+    finalConfigForProject.dockerOptions.registryUrl
+  );
+  return {
+    newVersion,
+    logs,
+  };
+}
+
+async function promptForNewVersion(
+  versionSchemes: Record<string, string>,
+  projectName: string
+) {
+  const { versionScheme } = await prompt<{ versionScheme: string }>({
+    name: 'versionScheme',
+    type: 'select',
+    message: `What type of release would you like to make for project "${projectName}"?`,
+    choices: Object.keys(versionSchemes).map((vs) => ({
+      name: vs,
+      message: vs,
+      value: vs,
+      hint: interpolateVersionPattern(versionSchemes[vs], { projectName }),
+    })),
+  });
+
+  return versionScheme;
+}
+
+function calculateNewVersion(
+  projectName: string,
+  versionScheme: string,
+  versionSchemes: Record<string, string>
+): string {
+  if (!(versionScheme in versionSchemes)) {
+    throw new Error(
+      `Could not find version scheme '${versionScheme}'. Available options are: ${Object.keys(
+        versionSchemes
+      ).join(', ')}.`
+    );
+  }
+  return interpolateVersionPattern(versionSchemes[versionScheme], {
+    projectName,
+  });
+}
+
+function updateProjectVersion(
+  newVersion: string,
+  workspaceRoot: string,
+  projectRoot: string,
+  repositoryName?: string,
+  registry?: string
+): string[] {
+  const isDryRun = process.env.NX_DRY_RUN && process.env.NX_DRY_RUN !== 'false';
```

Comment:
Why not

```suggestion
  const isDryRun = process.env.NX_DRY_RUN === 'true';
```

Reply:
I just matched what was used in other areas